### ISSUE TITLE
trace.overwrite and report.overwrite errors are reported differently

### ISSUE THREAD

**Issue by SamStudio8** [nextflow-io/nextflow#3627]
### Behavior

Since Nextflow 22.10, Nextflow is supposedly configured to cowardly refuse to start a workflow that would overwrite an existing `report`, `timeline` or `trace` unless the relevant overwrite option has been specified in the `nextflow.config`. However, the behaviour is different depending on the file that would be overwritten. For a trace, the workflow will refuse to start; for a report or timeline, the workflow will execute and refuse to overwrite the files at the end, only emitting a warning.

#### Report (and timeline)

```
// nextflow.config
report {
    enabled = true
    file = "report"
}
```

```
% touch report
% NXF_VER=22.12.0-edge nextflow run main.nf
N E X T F L O W  ~  version 22.12.0-edge
Launching `main.nf` [distraught_einstein] DSL2 - revision: 12613ffe0c
executor >  local (3)
[db/7dcd14] process > hoot (1) [100%] 3 of 3 ✔
WARN: Task runtime metrics are not reported when using macOS without a container engine
WARN: Failed to render execution report -- see the log file for details
```

```
# nextflow.log
[...]
nextflow.exception.AbortOperationException: Report file already exists:
[...]/report -- enable the 'report.overwrite' option in your config file to overwrite existing files
```
([...] removed for brevity)

#### Trace

```
// nextflow.config
trace {
    enabled = true
    file = "trace"
}
```

```
% touch trace
% NXF_VER=22.12.0-edge nextflow run main.nf
N E X T F L O W  ~  version 22.12.0-edge
Launching `main.nf` [deadly_bartik] DSL2 - revision: 12613ffe0c
Trace file already exists: [...]/trace -- enable the 'trace.overwrite' option in your config file to overwrite existing files
```

### Environment 

* Nextflow version: 22.12.0-edge
* Java version: OpenJDK Runtime Environment Homebrew (build 17.0.3+0)
* Operating system: macOS
* Bash version: zsh 5.8

### Additional context

This inconsistency is surprising and I'd expect Nextflow to have the same behaviour for all reports. Moreover the behaviour for `report` and `timeline` is quite annoying, as one could run a workflow for hours, only to have the report and timeline not be generated.

**Comment by bentsherman** [comment]
That is unfortunate. The reason is that the trace file is created at the beginning and updated as each task completes, whereas the report and timeline are rendered at the end.

**Comment by pditommaso** [comment]
It makes sense to have the same behaviour and stop the execution! Solved via b238d7e2.
